src/content/docs/mobile-monitoring/new-relic-mobile/mobile-sdk/stop-interaction.mdx

```diff
@@ -0,0 +1,622 @@
+metaDescription: Mobile app monitoring API used to stop a manually tracked interaction.
+redirects:
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/endinteraction-android-agent-api
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/endinteraction-android-sdk-api
+  - /docs/mobile-monitoring/new-relic-mobile-android/android-sdk-api/end-interaction
+  - /mobile-monitoring/new-relic-mobile-ios/ios-sdk-api/stop-current-interaction-ios-sdk-api
+  - /docs/mobile-monitoring/new-relic-mobile-ios/ios-sdk-api/stop-current-interaction
+---
+
+<Tabs>
+	<TabsBar>
+        <TabsBarItem id="android">
+            Android
+        </TabsBarItem>
+        <TabsBarItem id="ios">
+            iOS
+        </TabsBarItem>
+        <TabsBarItem id="capacitor">
+            Capacitor
+        </TabsBarItem>
+        <TabsBarItem id="cordova">
+            Cordova
+        </TabsBarItem>
+        <TabsBarItem id="flutter">
+            Flutter
+        </TabsBarItem>
+        <TabsBarItem id="react">
+            React Native
+        </TabsBarItem>
+        <TabsBarItem id="xamarin">
+            Xamarin
+        </TabsBarItem>
+        <TabsBarItem id="maui">
+            .NET MAUI
+        </TabsBarItem>
+    </TabsBar>
+
+    <TabsPages>
+        <TabsPageItem id="android">
+## Syntax [#syntax]
+
+### Java [#java] 
+
+```java
+NewRelic.endInteraction(string $interactionID)
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+NewRelic. endInteraction(id : String!)
+```
+
+## Description [#description]
+
+New Relic ends all interactions automatically, but you can use `endInteraction()` to end a custom interaction early. The string ID is returned when you use the `startInteraction()` call.
+
+This call has no effect if the interaction has already ended.
+
+## Parameters [#parameters]
+
+<table>
+  <thead>
+    <tr>
+      <th width={200}>
+        Parameter
+      </th>
+      <th width={200}>
+        Type
+      </th>
+      <th>
+        Description
+      </th>
+    </tr>
+  </thead>
+
+  <tbody>
+    <tr>
+      <td>
+        `$interactionID`
+      </td>
+
+            <td>
+        `string`
+      </td>
+
+      <td>
+        Required. The value returned by `startInteraction`.
+        It is required to pass this string to manually complete the interaction trace.
+      </td>
+    </tr>
+
+  </tbody>
+</table>
+
+## Example [#example]
+
+Here's an example of ending a custom interaction `RefreshContacts`:
+
+### Java [#java] 
+
+```java
+public class MainActivity extends Activity {
+  ...
+  @Override
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case R.id.action_refresh:
+        String interactionId = NewRelic.startInteraction("RefreshContacts");
+        ... 
+        return true;
+      default:
+        NewRelic.endInteraction(interactionId);
+        return super.onOptionsItemSelected(item);
+    }
+  }
+  ...
+}
+```
+
+### Kotlin [#kotlin]
+
+```kotlin
+
+```
```

Comment:
@ndesai-newrelic Can you please provide a kotlin example equivalent to the Java example?

Reply:
class MainActivity : AppCompatActivity() {

  .....
    var client: OkHttpClient = OkHttpClient();


        binding.fab.setOnClickListener { view ->

            val interActionId = NewRelic.startInteraction("Getting Data From Server")

                lifecycleScope.launch(Dispatchers.IO) {
                    val result = getRequest()
                    NewRelic.endInteraction(interActionId)
                }
            }
        }
    ....
    }

Reply:
```suggestion
```kotlin

class MainActivity : AppCompatActivity() {

.....
var client: OkHttpClient = OkHttpClient();

    binding.fab.setOnClickListener { view ->

        val interActionId = NewRelic.startInteraction("Getting Data From Server")

            lifecycleScope.launch(Dispatchers.IO) {
                val result = getRequest()
                NewRelic.endInteraction(interActionId)
            }
        }
    }
....
}

```
```